internal/controller/promotionstrategy_controller.go

```diff
@@ -303,62 +290,44 @@ func (r *PromotionStrategyReconciler) copyCommitStatuses(ctx context.Context, cs
 				continue
 			}
 
-			cs := promoterv1alpha1.CommitStatus{}
+			copiedCommitStatus := &promoterv1alpha1.CommitStatus{}
 			//TODO: do we like this name proposed-<name>?
-			copiedCSName := utils.KubeSafeUniqueName(ctx, "proposed-"+commitStatus.Name)
+			copiedCSName := utils.KubeSafeUniqueName(ctx, promoterv1alpha1.CopiedProposedCommitPrefixName+commitStatus.Name)
 			proposedCSObjectKey := client.ObjectKey{Namespace: commitStatus.Namespace, Name: copiedCSName}
-			errGet := r.Get(ctx, proposedCSObjectKey, &cs)
-			if errGet != nil {
-				if errors.IsNotFound(errGet) {
-					status := &promoterv1alpha1.CommitStatus{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:        proposedCSObjectKey.Name,
-							Annotations: commitStatus.Annotations,
-							Labels:      commitStatus.Labels,
-							Namespace:   commitStatus.Namespace,
-						},
-						Spec: promoterv1alpha1.CommitStatusSpec{
-							RepositoryReference: commitStatus.Spec.RepositoryReference,
-							Sha:                 copyToProposedHydratedSha,
-							Name:                branch + " - " + commitStatus.Spec.Name,
-							Description:         commitStatus.Spec.Description,
-							Phase:               commitStatus.Spec.Phase,
-							Url:                 "https://github.com/" + commitStatus.Spec.RepositoryReference.Owner + "/" + commitStatus.Spec.RepositoryReference.Name + "/commit/" + copyFromActiveHydratedSha,
-						},
-					}
-					if status.Labels == nil {
-						status.Labels = make(map[string]string)
-					}
-					status.Labels[promoterv1alpha1.CommitStatusLabelCopy] = "true"
-					status.Labels["promoter.argoproj.io/commit-status-copy-from"] = utils.KubeSafeLabel(ctx, commitStatus.Spec.Name)
-					status.Labels["promoter.argoproj.io/commit-status-copy-from-sha"] = utils.KubeSafeLabel(ctx, copyFromActiveHydratedSha)
-					status.Labels["promoter.argoproj.io/commit-status-copy-from-branch"] = utils.KubeSafeLabel(ctx, branch)
-					err := r.Create(ctx, status)
-					if err != nil {
-						return err
-					}
-				} else {
-					logger.Error(errGet, "failed to get CommitStatus", "namespace", proposedCSObjectKey.Namespace, "name", proposedCSObjectKey.Name)
-					return errGet
-				}
-			}
-			commitStatus.Spec.DeepCopyInto(&cs.Spec)
 
-			if cs.Labels == nil {
-				cs.Labels = make(map[string]string)
+			copiedCommitStatus = &promoterv1alpha1.CommitStatus{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        proposedCSObjectKey.Name,
+					Annotations: commitStatus.Annotations,
+					Labels:      commitStatus.Labels,
+					Namespace:   commitStatus.Namespace,
+				},
```

Comment:
Should we own this CS?

Reply:
I don't think so, it would be weird in ArgoCD to have this copied one show up but not the parent.

Reply:
Maybe the parent is the owner? Haven't thought it through, but maybe.

Reply:
Or maybe the ProposedCommit is the owner, same as the parent CommitStatus?

Reply:
Yea the CommitStatus maybe makes sense, idk depends on if we want to expose that detail to end users that use ArgoCD and have commit statuses checked in 🤷

Reply:
Fine to leave it for later